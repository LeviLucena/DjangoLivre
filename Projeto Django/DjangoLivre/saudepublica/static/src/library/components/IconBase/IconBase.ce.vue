<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
