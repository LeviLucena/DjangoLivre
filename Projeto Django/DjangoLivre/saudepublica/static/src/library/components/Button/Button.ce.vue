<template>
  <button
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    v-bind="$attrs"
    class="br-button"
    :class="{
      primary: type === 'primary',
      secondary: type === 'secondary',
      large: density === 'large',
      middle: density === 'middle',
      small: density === 'small',
      xsmall: density === 'xsmall',
      loading,
      inverted,
      active,
      circle,
      block,
    }"
    aria-label=""
    :type="submit ? 'submit' : 'button'"
    :disabled="disabled"
  >
    <slot>
      <IconBase v-if="icon" :icon-name="icon" :class-name="getShowCircle"></IconBase>
      <span v-if="!circle">
        <slot>{{ label }}</slot>
      </span>
    </slot>
  </button>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"
import IconBase from "../IconBase/IconBase.ce.vue"

const Button = {
  name: "brButton",
  components: {
    IconBase,
  },
  mixins: [TooltipMixin],
  props: {
    /**
     * **[OBRIGATÓRIO]** O rótulo é um elemento obrigatório. A **exceção** ocorre na versão tipo circular uma vez que não existe rótulo.
     */
    label: {
      type: String,
      default: null,
    },
    /**
     * **[OPCIONAL]** Ênfase do botão, com finalidade de proporcionar hierarquia visual a esses elementos e ações. Os botões mudam de aparência conforme a ênfase utilizada. A ênfase terciária é padrão. **[ primary | secondary ]**".
     */
    type: {
      type: String,
      validator: function (value) {
        return ["secondary", "primary", "tertiary"].includes(value)
      },
      default: null,
    },
    /**
     * **[Opcional]** Indica que o botão é do tipo *submit*.
     */
    submit: {
      type: Boolean,
      default: null,
    },
    /**
     * **[Opcional]** Tipo de botão muito usado dentro de outros componentes ou em situações onde a representação do ícone é suficiente para transmitir a mensagem ao usuário.
     */
    circle: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** O ajuste da densidade consiste em aumentar ou reduzir os espaçamentos internos do botão de forma a deixá-los mais compactos ou mais expandidos em seu formato. **[ large | middle | small ]**
     */
    density: {
      type: String,
      default: "middle",
      validator: function (value) {
        return ["large", "middle", "small", "xsmall"].includes(value)
      },
    },
    /** **[OPCIONAL]** Desabilita o botão */
    disabled: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** Apresenta o botão em comportamento de bloco. o botão “empurra” qualquer outro elemento existente para outro espaço disponível. */
    block: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Usado quando houver uma necessidade de enfatizar visualmente a ideia da ação. **Nos botões do tipo circular a utilização é obrigatória, visto que a mensagem da ação é transmitida apenas pelo ícone.**
     */
    icon: {
      type: [String],
      default: null,
    },
    /**
     * **[OPCIONAL]** Utilizado quando o background possui cores escuras.
     */
    inverted: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Utilizado para enfatizar o estado ativo do botão
     */
    active: {
      type: Boolean,
      default: false,
    },
    /**
     * **[OPCIONAL]** Aplica carregando
     */
    loading: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    getShowCircle() {
      return this.circle ? "" : "mr-1"
    },
  },
}

export default Button
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/button/_button";
@import "~@govbr-ds/core/src/components/loading/_loading";
@import "~@govbr-ds/core/src/components/notification/_notification";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
