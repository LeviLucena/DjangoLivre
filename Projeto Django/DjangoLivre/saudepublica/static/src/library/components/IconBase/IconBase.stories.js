import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
