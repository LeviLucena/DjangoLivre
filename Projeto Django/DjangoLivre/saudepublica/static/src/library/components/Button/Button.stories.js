import { kebabiseArgs } from '../../util/Utils.js'
import Button from './Button.ce.vue'

export default {
  title: 'Componentes/Button',
  component: Button,
  parameters: {
    backgrounds: {
      default: 'light',
      values: [
        { name: 'light', value: '#fff' },
        { name: 'dark', value: '#071D41' },
      ],
    },
    actions: {
      handles: ['click'],
    },
  },
  argTypes: {
    label: {
      control: 'text',
    },
    type: {
      type: 'select',
      options: ['secondary', 'primary'],
    },
    density: {
      type: 'select',
      options: ['large', 'middle', 'small', 'xsmall'],
    },
    circle: {
      defaultValue: false,
    },
    disabled: {
      defaultValue: false,
    },
    block: {
      defaultValue: false,
    },
    icon: {
      defaultValue: '',
    },
    inverted: {
      defaultValue: false,
    },
    active: {
      defaultValue: false,
    },
    loading: {
      defaultValue: false,
    },
    tooltipText: {
      control: 'text',
    },
    tooltipPlace: {
      options: ['top', 'right', 'bottom', 'left'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipType: {
      options: ['info', 'success', 'warning', 'error'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipTimer: {
      control: 'number',
      if: { arg: 'tooltipText', truthy: true },
    },
    default: {
      control: 'text',
      description: 'Rótulo do botão, que pode ser passado por slot',
      table: {
        type: {
          summary: `string`,
        },
        defaultValue: {
          summary: '',
        },
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-button v-bind="args"></br-button>`,
})

const TemplateExternal = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-button v-bind="args">
  <br-icon-base icon-name="user"></br-icon-base>
  </br-button>`,
})

export const Primary = Template.bind({})
Primary.args = {
  label: 'Primary',
  type: 'primary',
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Secondary',
  type: 'secondary',
}

export const Tertiary = Template.bind({})
Tertiary.args = {
  label: 'Tertiary',
  inverted: false,
}

export const Circle = Template.bind({})
Circle.parameters = { controls: { exclude: ['label', 'block'] } }
Circle.args = {
  type: 'primary',
  circle: true,
  icon: 'trash',
  inverted: false,
}

export const Loading = Template.bind({})
Loading.parameters = {
  controls: { exclude: ['label', 'block', 'disabled', 'icon', 'circle'] },
}
Loading.args = {
  type: 'secondary',
  loading: true,
}

export const Inverted = Template.bind({})
Inverted.parameters = { backgrounds: { default: 'dark' } }
Inverted.args = {
  label: 'Inverted',
  type: 'secondary',
  inverted: true,
}

export const IconAndLabel = Template.bind({})
IconAndLabel.args = {
  label: 'Text',
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const IconAndCircle = Template.bind({})
IconAndCircle.parameters = { controls: { exclude: ['label', 'block'] } }
IconAndCircle.args = {
  circle: true,
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const IconAndCircleExternal = TemplateExternal.bind({})
IconAndCircleExternal.parameters = { controls: { exclude: ['label', 'block'] } }
IconAndCircleExternal.args = {
  circle: true,
  type: 'secondary',
  icon: 'city',
  inverted: false,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  label: 'Botão com tooltip',
  type: 'primary',
  tooltipText: 'Lorem ipsum dolor sit amet.',
  tooltipType: 'warning',
}
