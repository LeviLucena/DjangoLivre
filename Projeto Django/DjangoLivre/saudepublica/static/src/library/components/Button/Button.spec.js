import { shallowMount } from '@vue/test-utils'
import IconBase from '../IconBase/IconBase.ce.vue'
import Button from './Button.ce.vue'

const densidades = ['large', 'middle', 'small']
const tipo = ['secondary', 'primary']

describe('Button', () => {
  test('it renders icon', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        icon: 'home',
      },
      global: {
        stubs: {
          'icon-base': IconBase,
        },
      },
    })
    expect(wrapper.props().icon).toBe('home')
  })

  test('it renders br-button', () => {
    const wrapper = shallowMount(Button)
    expect(wrapper.classes('br-button')).toBe(true)
  })

  test('set label attribute', () => {
    const rotulo = 'Label do button'
    const wrapper = shallowMount(Button, {
      propsData: {
        label: rotulo,
      },
    })
    expect(wrapper.text()).toMatch(rotulo)
  })

  test('set circle attribute', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        circle: true,
      },
    })
    expect(wrapper.classes('circle')).toBe(true)
  })

  test('assign inverted property to button and check if it is true', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        inverted: true,
      },
    })
    expect(wrapper.find('.br-button.inverted').exists()).toBe(true)
  })

  test('set disabled attribute', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        disabled: true,
      },
    })
    expect(wrapper.find('button[disabled]').exists()).toBe(true)
  })

  test('set block attribute', () => {
    const wrapper = shallowMount(Button, {
      propsData: {
        block: true,
      },
    })
    expect(wrapper.classes('block')).toBe(true)
  })

  densidades.forEach((densidade) => {
    test(`set density attribute ${densidade}`, () => {
      const wrapper = shallowMount(Button, {
        propsData: {
          density: densidade,
        },
      })
      expect(wrapper.find(`.${densidade}`).exists()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  tipo.forEach((cor) => {
    test(`set type attribute ${cor}`, () => {
      const wrapper = shallowMount(Button, {
        propsData: {
          type: cor,
        },
      })
      expect(wrapper.find(`.${cor}`).exists()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
