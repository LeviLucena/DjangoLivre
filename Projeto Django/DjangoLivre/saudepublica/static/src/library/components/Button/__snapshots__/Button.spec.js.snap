// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button set density attribute large 1`] = `
<button
  aria-label=""
  class="br-button large"
  type="button"
>
  
  <!--v-if-->
  <span>
    
    
    
  </span>
  
</button>
`;

exports[`Button set density attribute middle 1`] = `
<button
  aria-label=""
  class="br-button middle"
  type="button"
>
  
  <!--v-if-->
  <span>
    
    
    
  </span>
  
</button>
`;

exports[`Button set density attribute small 1`] = `
<button
  aria-label=""
  class="br-button small"
  type="button"
>
  
  <!--v-if-->
  <span>
    
    
    
  </span>
  
</button>
`;

exports[`Button set type attribute primary 1`] = `
<button
  aria-label=""
  class="br-button primary middle"
  type="button"
>
  
  <!--v-if-->
  <span>
    
    
    
  </span>
  
</button>
`;

exports[`Button set type attribute secondary 1`] = `
<button
  aria-label=""
  class="br-button secondary middle"
  type="button"
>
  
  <!--v-if-->
  <span>
    
    
    
  </span>
  
</button>
`;
